pkg/skaffold/build/local/local_test.go

```diff
@@ -129,12 +187,33 @@ func TestLocalRun(t *testing.T) {
 			api: testutil.FakeAPIClient{
 				ErrImageBuild: true,
 			},
-			shouldErr: true,
+			expectedResults: []testResult{
+				{
+					buildResult: build.Result{
+						Target: latest.Artifact{
+							ImageName: "gcr.io/test/image",
+							ArtifactType: latest.ArtifactType{
+								DockerArtifact: &latest.DockerArtifact{},
+							},
+						},
+						Error: errors.New("building [gcr.io/test/image]"),
+					},
+					shouldErr: true,
+				},
+			},
 		},
 		{
-			description: "unkown artifact type",
+			description: "unknown artifact type",
```

Comment:
hmmm...with the explicit error checking this kind of shows that this is a weird error message, especially with the space at the end. Can we improve it here a little bit now that we ran into it? 

This is not testing the unknown artifact type branch in the local builder but that an error is returned from the InSequence builder that the tags don't contain the image name - which is also empty.... **very** confusing. 

This doesn't work but shows the direction we should go:

```
{
			description: "unknown artifact type",
			artifacts: []*latest.Artifact{{
				ImageName:    "gcr.io/test/image",
				ArtifactType: latest.ArtifactType{}},
			},
			tags: tag.ImageTags(map[string]string{"gcr.io/test/image": "gcr.io/test/image:tag"}),
			expectedResults: []testResult{
				{
					buildResult: build.Result{
						Target: latest.Artifact{},
						Error:  errors.New("undefined artifact type: nil"),
					},
					shouldErr: true,
				},
			},
		},
```

Reply:
I don't disagree with your suggestion but I don't think the is the right place to make the change. this will require adding some new logic in the way that we retrieve the tags for a parsed artifact, and conditionally returning a different error. there's already a lot going on here, so I want to keep the scope as minimal as possible. I know that touching all this code is exposing lots of little problems, but let's address these in a follow up PR.

Reply:
:+1: Sounds good. Can you add an issue to have this fixed?